src/main/webapp/2Dscript.js

```diff
@@ -80,7 +175,418 @@ function addCenterControl(controlDiv) {
+ */
+function placeRectangleStart() {
+  x = event.pageX;
+  y = event.pageY;
+  priorLat = currentLat;
+  priorLng = currentLng;
+  $('body').append('<div class="widget" style="top:' +
+    y + 'px; left: ' + x + 'px;"></div>');
+  widget = $('.widget').last();
+  isMouseDown = true;
+}
+
+/**
+ * Finish placing the selection box, select the subsection within this region.
+ */
+function placeRectangleEnd() {
+  $('.widget').remove();
+  // Clear pop up window.
+  if (infoWindow !== undefined) {
+    infoWindow.setMap(null);
+  }
+  // Clear other paths.
+  // line
+  // minLatPoint
+  // lat , lng
+  // minLngPoint
+  // lat, lng
+  const subSectionData = computeSubSection(currentRun.data,
+      currentLat, priorLat, currentLng, priorLng);
+  // Choose a subSectionNumber, implement differently in future pr.
+  const subSectionNumber = 1;
+  // Clear prior paths, only display newly selected ones.
+  if (markerBottom !== undefined) {
+    markerBottom.setMap(null);
+  }
+  if (markerTop !== undefined) {
+    markerTop.setMap(null);
+  }
+  if (subPath !== undefined) {
+    subPath.setMap(null);
+  }
+  subPath = getPolyLine(subLine, 'blue', 1.0, 2);
+  // Setup event listener to show option for 3D window when polyline is
+  // clicked.
+  google.maps.event.addListener(subPath, 'click', linkTo3D);
+  subPath.setMap(map);
+  markerBottom = genMarker(subSectionData.minLatPoint.lat,
+      subSectionData.minLatPoint.lng);
+  markerTop = genMarker(subSectionData.maxLngPoint.lat,
+      subSectionData.maxLngPoint.lng);
+  markerBottom.setMap(map);
+  markerTop.setMap(map);
+
+  sessionStorage.setItem(id + '_' + type + '_' + subSectionNumber,
+      JSON.stringify(subLine));
+}
+
+/**
+ * Change the rectangle width to be match the current cursor position.
+ * @param {MouseEvent} event
+ */
+function genChangingBox(event) {
+  if (isMouseDown) {
+    console.log('Set width triggered');
+    finX = event.pageX;
+    finY = event.pageY;
+    widget.width(finX - x);
+    widget.height(finY - y);
+    widget.css({
+      'width': (finX - x) + '!important',
+      'height': (finY - y) + '!important',
+      'display': 'block',
+      'border': '2px dashed #ccc',
+    });
+  }
+}
+
+/**
+ * Generate a Google maps marker at the given lat and long.
+ * @param {number} latitude
+ * @param {number} longitude
+ * @return {google.maps.Marker}
+ */
+function genMarker(latitude, longitude) {
+  const marker = new google.maps.Marker({
+    position: {lat: latitude, lng: longitude},
+    title: 'Lat: ' + latitude + ' Lng: ' + longitude,
+  });
+  return marker;
+}
+
+
+/**
+ * Create a polyline from the given points with specified parameters.
+ * @param {Point} linePoints
+ * @param {string} color
+ * @param {number} opacity
+ * @param {number} weight
+ * @param {number} index
+ * @return {google.maps.Polyline}
+ */
+function getPolyLine(linePoints, color, opacity, weight, index = 1) {
+  const polyLine = new google.maps.Polyline({
```

Comment:
Follow the API's lead on capitalization: "polyline" is all one word, so the L shouldn't be capitalized.